docs/noencap-hybrid-modes.md

```diff
@@ -0,0 +1,150 @@
+option is defined in `antrea-agent.conf` of the `antrea` ConfigMap in the
+[Antrea deployment YAML](https://github.com/vmware-tanzu/antrea/blob/main/build/yamls/antrea.yml).
+
+```yaml
+  antrea-agent.conf: |
+    ... ...
+    trafficEncapMode: hybrid
+    ... ...
+```
+
+After changing the config option, you can deploy Antrea in `Hybrid` mode with
+the usual command:
+
+```bash
+kubectl apply -f antrea.yml
+```
+
+## NoEncap Mode
+
+In `NoEncap` mode, Antrea never encapsulates Pod traffic. Just like `Hybrid`
+mode, the Node network needs to allow Pod IP addresses sent out from Nodes. When
+the Nodes are not in the same subnet, `NoEncap` mode additionally requires the
+Node network be able to route the Pod traffic from the source Node to the
+destination Node. There are two possibilities to enable this routing by Node
+network:
+
+* Leverage Route Controller of [Kubernetes Cloud Controller Manager](https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller).
+The Kubernetes Cloud Providers that implement Route Controller can add routes
+to the cloud network routers for the Pod CIDRs of Nodes, and then the cloud
+network is able to route Pod traffic between Nodes. This Route Controller
+functionality is supported by the Cloud Provider implementations of the major
+clouds, including: [AWS](https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/legacy-cloud-providers/aws),
+[Azure](https://github.com/kubernetes-sigs/cloud-provider-azure),
+[GCE](https://github.com/kubernetes/kubernetes/tree/master/staging/src/k8s.io/legacy-cloud-providers/gce),
+and [vSphere (with NSX-T)](https://github.com/kubernetes/cloud-provider-vsphere).
+
+* Run a routing protocol or even manually configure routers to add routes to
+the Node network routers. For example, Antrea can work with [kube-router](https://www.kube-router.io)
+and leverage kube-router to advertise Pod CIDRs to routers using BGP. Section
+[Using kube-router for BGP](#using-kube-router-for-bgp) describes how to
+configure Antrea and kube-router to work together.
+
+When the Node network can support forwarding and routing of Pod traffic, Antrea
+can be configured to run in the `NoEncap` mode, by setting the `trafficEncapMode`
+config option of `antrea-agent` to `noEncap`. By default, Antrea performs SNAT
+(source network address translation) for the outbound connections from a Pod to
+outside of the Pod network, using the Node's IP address as the SNAT IP. In the
+`NoEncap` mode, as the Node network knows about Pod IP addresses, the SNAT by
+Antrea might be unnecessary. In this case, you can disable it by setting the
+`noSNAT` config option to `true`. The `trafficEncapMode` and `noSNAT` config
+options are defined in `antrea-agent.conf` of the `antrea` ConfigMap in the
+[Antrea deployment YAML](https://github.com/vmware-tanzu/antrea/blob/main/build/yamls/antrea.yml).
+
+```yaml
+  antrea-agent.conf: |
+    ... ...
+    trafficEncapMode: noEncap
+
+    #noSNAT: false
```

Comment:
I think you mean `noSNAT: true`?

Reply:
I wanted to show the default value, but yes it seems confusing. Changed to: noSNAT: false.